File: PathStructureClient/src/dtos/PathChangedNotificationParams.js
const { PathMatchDto } = require('./PathMatchDto');

class PathChangedNotificationParams {
  /**
   * @param {object} options
   * @param {string} options.message
   * @param {string} options.path
   * @param {PathMatchDto|null} options.currentMatch
   * @param {Record<string, string>} options.variables
   * @param {PathMatchDto[]} options.matches
   * @param {PathMatchDto[]} options.immediateChildMatches
   * @param {string} options.timestamp
   */
  constructor({
    message,
    path,
    currentMatch,
    variables,
    matches,
    immediateChildMatches,
    timestamp
  }) {
    /** @type {string} */
    this.message = message;
    /** @type {string} */
    this.path = path;
    /** @type {PathMatchDto|null} */
    this.currentMatch = currentMatch;
    /** @type {Record<string, string>} */
    this.variables = variables;
    /** @type {PathMatchDto[]} */
    this.matches = matches;
    /** @type {PathMatchDto[]} */
    this.immediateChildMatches = immediateChildMatches;
    /** @type {string} */
    this.timestamp = timestamp;
  }
}

module.exports = { PathChangedNotificationParams };

File: PathStructure.WatcherHost/Dtos/PathChangedNotificationParams.cs
using System.Collections.Generic;

namespace PathStructure.WatcherHost.Dtos
{
    /// <summary>
    /// Represents path change notification parameters.
    /// </summary>
    internal sealed class PathChangedNotificationParams
    {
        /// <summary>
        /// Gets or sets the notification message.
        /// </summary>
        public string Message { get; set; }

        /// <summary>
        /// Gets or sets the raw Explorer path.
        /// </summary>
        public string Path { get; set; }

        /// <summary>
        /// Gets or sets the current matched path structure.
        /// </summary>
        public PathMatchDto CurrentMatch { get; set; }

        /// <summary>
        /// Gets or sets the resolved template variables.
        /// </summary>
        public IReadOnlyDictionary<string, string> Variables { get; set; }

        /// <summary>
        /// Gets or sets the full match trail for the selection.
        /// </summary>
        public List<PathMatchDto> Matches { get; set; }

        /// <summary>
        /// Gets or sets the immediate child matches.
        /// </summary>
        public List<PathMatchDto> ImmediateChildMatches { get; set; }

        /// <summary>
        /// Gets or sets the ISO timestamp for the change event.
        /// </summary>
        public string Timestamp { get; set; }
    }
}
